src/entities/ticket.ts

```diff
@@ -27,3 +28,63 @@ export function getTicket(eventIndex: BigInt, tokenId: BigInt): Ticket {
 
   return ticket as Ticket;
 }
+
+function calculateReservedFuel(
+  basePrice: BigDecimal,
+  minFee: BigDecimal,
+  maxFee: BigDecimal,
+  rate: BigDecimal,
+  price: BigDecimal
+): BigDecimal {
+  // The reserved fuel per-ticket is not emitted in the events for efficiency. Here we replicate the calculation made
+  // in Economics.sol for the reservedFuel on the Ticket entity. Minute differences may occur due to using BigDecimal
+  // rather than BigInt. There should be no differences on other entities as they use the reservedFuel from the event.
+  //
+  // From get-protocol-core: https://github.com/GETProtocol/get-protocol-core/blob/release/contracts/Economics.sol#L179
+  //
+  // uint256 _fuelUsd = uint256(_ticketActions[i].basePrice) * _product.rate * 1e9;
+  // if (_fuelUsd < _min) {
+  //     _fuel += (_min * 1e18) / integrator.price;
+  // } else if (_fuelUsd > _max && _max != 0) {
+  //     _fuel += (_max * 1e18) / integrator.price;
+  // } else {
+  //     _fuel += (_fuelUsd * 1e18) / integrator.price;
+  // }
+
+  let fuelUsd = basePrice.times(rate).div(BigDecimal.fromString("100"));
```

Comment:
Struggled to understand why the division was by 1e2 only until I saw that the rate was already divided by 1e4 here: https://github.com/GETProtocol/subgraph/blob/d752441998fbc7c8cf6af58c56c184118fca2efb/src/mappings/v2/economics.ts#L47

Reply:
Ah nice catch, yes it's not completely clear admittedly!